### DISCUSSION TITLE
Can we temporarily disable 'Add' button for WebAuthn after clicking to prevent sending multiple verification emails

### DISCUSSION THREAD

**Discussion by yusiwen** [authelia/authelia#7516] [Ideas]
After clicking the 'Add' button for WebAuthn, the server will send a verification email to the user. But use still can click the Add button when the processing is ongoing. So if the mail server or the network is not very responsive, user may click the Add button multiple times, causing the server to send a lot emails. It happened to me when I first add my WebAuthn credential: I clicked the button three times before the dialog for me to input verification code shown up, and I got three those emails.

For better user experiences, can we temporarily disable this 'Add' button after clicking, and show a progress cycle indicates that the server is working to send mails?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Yes, it is possible to improve the user experience by temporarily disabling the 'Add' button after it is clicked and showing a progress indicator. This would prevent multiple clicks and the subsequent sending of multiple emails. This suggestion aligns with best practices for user interface design, ensuring that users are aware that their action is being processed and preventing redundant actions.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Give the `fix-add-credential-ux` docker tag a try.

**Reply by james-d-elliott** [reply]
Any feedback @yusiwen ?

**Reply by yusiwen** [reply]
I will give it a try ASAP, sorry for the late reply

**Reply by yusiwen** [reply]
@james-d-elliott 
I've tested it. it works like a charm! love it! thanks a lot! 
👍

**Reply by james-d-elliott** [reply]
Merging for 4.38.10, thanks for the feedback
